src/lib/SmartTokens/Types/PTokenDirectory.hs

```diff
@@ -188,8 +154,12 @@ isHeadNode = plam $ \node ->
 
 isTailNode :: ClosedTerm (PAsData PDirectorySetNode :--> PBool)
 isTailNode = plam $ \node ->
-  pfield @"next" # node #== pemptyCSData
+  pfield @"next" # node #== ptailNextData
```

Comment:
Bug 3. isTail should check that next is the tail currency symbol (highest possible lexographic CS).